pilot/pkg/security/model/authentication.go

```diff
@@ -58,6 +61,76 @@ const (
 	IngressGatewaySdsCaSuffix = "-cacert"
 )
 
+// WorkloadSdsConfig encapsulates all SDS-related configuration data
+type WorkloadSdsConfig struct {
+	UdsPath             string
+	SdsEnabled          bool
+	EnableSdsTokenMount bool
+	SdsUseK8sJwt        bool
+}
+type workloadSdsConfigBuilder struct {
+	udsPath             string
+	sdsEnabled          bool
+	enableSdsTokenMount bool
+	sdsUseK8sJwt        bool
+}
+
+// WorkloadSdsConfigBuilder provides sensible SDS defaults and validation
+type WorkloadSdsConfigBuilder interface {
+	SetEnabled(enabled bool) WorkloadSdsConfigBuilder
+	SetUdsPath(path string) WorkloadSdsConfigBuilder
+	SetEnableTokenMount(enabled bool) WorkloadSdsConfigBuilder
+	SetUseK8sJwt(enabled bool) WorkloadSdsConfigBuilder
+	Build() (WorkloadSdsConfig, error)
+}
+
+// NewWorkloadSdsConfigBuilder generates a new WorkloadSdsConfigBuilder
+func NewWorkloadSdsConfigBuilder() WorkloadSdsConfigBuilder {
```

Comment:
Would it be better to return *WorkloadSdsConfigBuilder here and below? It seems that the methods below make a copy of the struct and return that copy.

Reply:
Agreed, returning ref makes more sense.

Do you think the builder pattern as a whole has any merit here, or should this just be boiled down to a struct + validation func?